modules/afamqp/afamqp.c

```diff
@@ -342,8 +384,41 @@ afamqp_dd_connect(AMQPDestDriver *self, gboolean reconnect)
       msg_error("Error allocating AMQP connection.");
       goto exception_amqp_dd_connect_failed_init;
     }
+  if(strcmp(self->ca_file,"") != 0)
```

Comment:
it would be great if you could extract this large chunk of code to a separate function, one that is meant to create the amqp socket based on various settings.

Also, I don't really like ca_file being the trigger for setting up SSL. what if the user disables peer-verify (I know it's bad and why it is bad, but it happens), and they couldn't enable SSL in any way.

I have two ideas how to do this:

 * in the tcp() destination, we use a tls() block, where all the tls specific options are written. TLS is enabled with the presence of the block, so this enables TLS:

tcp('1.2.3.4' tls())

often, it looks like this:
tcp('1.2.3.4' tls(key-file('aaa.aaa')));

* another idea is to use a tls(yes/no) option

My preference goes to
1) tls block
2) tls(yes/no) option for enabling it specifically

But let me ask @fekete-robert from the Balabit documentation team if he has a better idea